tiledb/sm/c_api/tiledb.h

```diff
@@ -3645,6 +3645,103 @@ TILEDB_EXPORT int32_t tiledb_query_set_buffer_var_nullable(
     uint8_t* buffer_validity_bytemap,
     uint64_t* buffer_validity_bytemap_size);
 
+/**
+ * Sets the buffer for a fixed-sized attribute/dimension to a query, which will
+ * either hold the values to be written (if it is a write query), or will hold
+ * the results from a read query.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * int32_t a1[100];
+ * uint64_t a1_size = sizeof(a1);
+ * tiledb_query_set_data_buffer(ctx, query, "a1", a1, &a1_size);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param query The TileDB query.
+ * @param name The attribute/dimension to set the buffer for. Note that
+ *     zipped coordinates have special name `TILEDB_COORDS`.
```

Comment:
It may be worthwhile to have this note in a separate `@note` after the `@return` to make it more obvious and reduce clutter in the `@param`
(Same in other changed APIs below)